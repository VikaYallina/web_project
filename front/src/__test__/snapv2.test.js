import React from 'react';
import * as redux from 'react-redux'
import {Home} from '../components/home.component'
import renderer from 'react-test-renderer';
import "core-js/stable";
import "regenerator-runtime/runtime";

const spy = jest.spyOn(redux, 'useSelector')
spy.mockReturnValue({ username:'test' })

//Use array destructurig to create mock functions.
let [getVideos, addToCart] = new Array(2).fill(jest.fn());

const props = {
  getVideos: getVideos,
  isAuthenticated: true,
  addToCart: addToCart,
  item: {
    items: new Array(1).fill({id:1, title:"Test", price:12.12, category:"sport", description:"wfwfwfwfw"}), 
    loading: false},
  user:   {
    accessToken: "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJ1c2VybmFtZSIsImlhdCI6MTYzOTQ3NzQ5NiwiZXhwIjoxNjM5NTYzODk2fQ.52Aky798yCUOTNBvKzsXybLjdOdSgk9qbtoT3ABBVpUaTSlM7LPvsEjCprU7UICGjIjOBwsGOJm55Clmsowr7w",
    email: "hotl@mial.ru",
    id: 7,
    roles: [],
    tokenType: "Bearer",
    username: "username"
  }
  
}

describe('Shallow rendered Todo Card', () =>{
    it('Testst', () => {
    const tree = renderer.create(
        <Home {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
    })
})