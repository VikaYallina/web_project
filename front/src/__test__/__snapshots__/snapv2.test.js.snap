// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendered Todo Card Testst 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4"
      >
        <div
          className="mb-4 card"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Test
            </h5>
            <h6
              className="card-subtitle"
            >
              Rs. 
              12.12
            </h6>
            <p
              className="card-text"
            >
              sport
            </p>
            <button
              aria-label={null}
              className="btn btn-success btn-sm"
              onClick={[Function]}
              type="button"
            >
              Add To Cart
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
